crates/ruff_linter/src/source_kind.rs

```diff
@@ -217,11 +217,49 @@ impl<'a> CodeDiff<'a> {
     }
 }
 
+trait ShowNonprinting {
+    fn show_nonprinting(&self) -> String;
+}
+
+impl ShowNonprinting for &str {
+    fn show_nonprinting(&self) -> String {
+        let mut output = String::with_capacity(self.len());
+
+        for byte in self.bytes() {
```

Comment:
Do I understand this correctly, that this will escape all non-ascii characters (in addition to some ascii characters)? I think that's undesirable. I would be very confused to see ä escaped in a Diff and wouldn't understand where this is coming from. 

I recommend focusing on the non printable characters and would consider using unicode characters for them (similar to what insta does see link above) or biomejs (https://github.com/MichaReiser/biome/blob/96bbf507a006274518e8f5ae5af57c56b7771fb0/crates/biome_diagnostics/src/display/frame.rs#L394-L406)

Reply:
Yes, that was correct. Fixed in the recent version.